torch/quantization/ns/utils.py

```diff
@@ -34,14 +35,17 @@ class NodeIOType(enum.Enum):
 def get_node_io_type(node: Node, gm: GraphModule) -> NodeIOType:
     if node.op == 'call_function':
         fp32_fun_target_names = ('torch.nn.functional', 'torch.nn')
+        # hack alert: this is not ready for production
+        # TODO(future PR): use a real mapping
+        fp32_funs = (torch.cat,)
```

Comment:
Doesnt torch.cat support both fp32 and quantized ops? My understanding is that the dispatcher handles dtype specific calls for the same function name.  For figuring out the i.o type, do you need to look at the dtypes of the inputs and outputs for a node?

Reply:
Yeah, the inside of this function is not ready for a wide variety for users, for sure.  There other hacks in here which are just as bad / worse :). I'm punting a better implementation of the insides of this function until the future.

We should align on the right interface though, as that is important to the design.  Currently, this is used to figure out what transformation, if any, we need to do to get from a node in B to the corresponding node in A.  This would be a dequant (the only thing supported now), an `nn.Identity()` (not in yet, but easy to add), a quant (might be harder to add and not a p0), a cast to fp16, etc.

We can probably rename this enum to `NodeFirstInputType` to make it clear, since that is the only thing used today.  If in the future we need types of other inputs, or outputs, we can change accordingly.